drivers/flash/flash_shell.c

```diff
@@ -255,6 +255,158 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
 	return result;
 }
 
+static int set_bypass(const struct shell *sh, shell_bypass_cb_t bypass)
+{
+	static bool in_use;
+
+	if (bypass && in_use) {
+		shell_error(sh, "flash load supports setting bypass on a single instance.");
+
+		return -EBUSY;
+	}
+
+	/* Mark that we have set or unset the bypass function */
+	in_use = bypass != NULL;
+
+	if (in_use) {
+		shell_print(sh, "Loading...");
+	}
+
+	shell_set_bypass(sh, bypass);
+
+	return 0;
+}
+
+#define FLASH_LOAD_BUF_MAX 256
+
+static const struct device *flash_load_dev;
+static uint32_t flash_load_addr;
+static uint32_t flash_load_total;
+static uint32_t flash_load_written;
+static uint32_t flash_load_chunk;
+
+static uint32_t flash_load_boff;
+static uint8_t flash_load_buf[FLASH_LOAD_BUF_MAX];
+
+static void bypass_cb(const struct shell *sh, uint8_t *recv, size_t len)
+{
+	uint32_t left_to_read = flash_load_total - flash_load_written - flash_load_boff;
+	uint32_t to_copy = MIN(len, left_to_read);
+	uint32_t copied = 0;
+
+	while (copied < to_copy) {
+
+		uint32_t buf_copy = MIN(to_copy, FLASH_LOAD_BUF_MAX - flash_load_boff);
+
+		memcpy(flash_load_buf + flash_load_boff, recv + copied, buf_copy);
+
+		flash_load_boff += buf_copy;
+		copied += buf_copy;
+
+		/* Buffer is full. Write data to memory. */
+		if (flash_load_boff == FLASH_LOAD_BUF_MAX) {
+			uint32_t addr = flash_load_addr + flash_load_written;
+			int rc = flash_write(flash_load_dev, addr, flash_load_buf,
+					FLASH_LOAD_BUF_MAX);
+
+			if (rc != 0) {
+				shell_error(sh, "Write to addr %x on dev %p ERROR!",
+						addr, flash_load_dev);
+			}
+
+			shell_print(sh, "Written chunk %d", flash_load_chunk);
+
+			flash_load_written += FLASH_LOAD_BUF_MAX;
+			flash_load_chunk++;
+			flash_load_boff = 0;
+		}
+	}
+
+	/* When data is not aligned to FLASH_LOAD_BUF_MAX there may be partial write
+	 * at the end.
+	 */
+	if (flash_load_written < flash_load_total &&
+			flash_load_written + flash_load_boff >= flash_load_total) {
+
+		uint32_t addr = flash_load_addr + flash_load_written;
+		int rc = flash_write(flash_load_dev, addr, flash_load_buf, flash_load_boff);
+
+		if (rc != 0) {
+			set_bypass(sh, NULL);
+			shell_error(sh, "Write to addr %x on dev %p ERROR!",
+					addr, flash_load_dev);
+			return;
+		}
+
+		shell_print(sh, "Written chunk %d", flash_load_chunk);
+		flash_load_written += flash_load_boff;
+		flash_load_chunk++;
+	}
+
+	if (flash_load_written >= flash_load_total) {
+		set_bypass(sh, NULL);
+		shell_print(sh, "Read all");
+	}
+}
+
+static int cmd_load(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *flash_dev;
+	int result;
+	uint32_t addr;
+	uint32_t size;
+
+	result = parse_helper(sh, &argc, &argv, &flash_dev, &addr);
+	if (result) {
+		return result;
+	}
+
+	size = strtoul(argv[2], NULL, 0);
+
+	/* Check if size is word aligned */
+	if ((size & (sizeof(void *) - 1)) != 0) {
```

Comment:
This limit might be dependent from the target flash memory.
Allowed alignment might differ form 32b (especially be higher).
You can use `flash_get_write_block_size()` for get this alignment.

Reply:
Done.